phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -2376,17 +2393,42 @@ public ResultSet executeQuery(String sql) throws SQLException {
 
     @Override
     public int executeUpdate(String sql) throws SQLException {
+        CompilableStatement stmt = preExecuteUpdate(sql);
+        int updateCount = executeMutation(stmt, createAuditQueryLogger(stmt, sql));
+        flushIfNecessary();
+        return updateCount;
+    }
+
+    /**
+     * Executes the given SQL statement similar to JDBC API executeUpdate() but also returns the
+     * updated or non-updated row as Result object back to the client. This must be used with
+     * auto-commit Connection. This makes the operation atomic.
+     * If the row is successfully updated, return the updated row, otherwise if the row
+     * cannot be updated, return non-updated row.
+     *
+     * @param sql The SQL DML statement, UPSERT or DELETE for Phoenix.
```

Comment:
How does it work for DELETE ? I can delete multiple rows using a single delete query. Does it handle that case ?

Reply:
DELETE support is coming as separate follow up Jira. I did not want to complicate this PR.

Reply:
But yes reg your question, since single DELETE can delete multiple rows, similar to UPSERT, we want to restrict returning value only if single row was deleted.